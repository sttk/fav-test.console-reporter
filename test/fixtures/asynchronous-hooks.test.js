'use strict';

function Connection() {
  this.saveError = null;
  this.clearError = null;
  this.findError = null;
  this.users = [];
}
Connection.prototype.save = function(users, cb) {
  this.users = users;
  cb(this.saveError);
};
Connection.prototype.clear = function(cb) {
  cb(this.clearError);
};
Connection.prototype.find = function(obj, cb) {
  cb(this.findError, this.users);
};

function User(name) {
  this.name = name;
}

var should = require('should');

describe('Connection', function() {
  var db = new Connection,
    tobi = new User('tobi'),
    loki = new User('loki'),
    jane = new User('jane');

  beforeEach(function(done) {
    db.clear(function(err) {
      if (err) return done(err);
      db.save([tobi, loki, jane], done);
    });
  });

  describe('#find()', function() {
    it('respond with matching records', function(done) {
      db.find({ type: 'User' }, function(err, res) {
        if (err) return done(err);
        res.should.have.length(3);
        done();
      });
    });
  });
});

describe('Connection (save error)', function() {
  var db = new Connection,
    tobi = new User('tobi'),
    loki = new User('loki'),
    jane = new User('jane');

  db.saveError = new Error('Save error');

  beforeEach(function(done) {
    db.clear(function(err) {
      if (err) return done(err);
      db.save([tobi, loki, jane], done);
    });
  });

  describe('#find()', function() {
    it('respond with matching records', function(done) {
      db.find({ type: 'User' }, function(err, res) {
        if (err) return done(err);
        res.should.have.length(3);
        done();
      });
    });
  });
});

