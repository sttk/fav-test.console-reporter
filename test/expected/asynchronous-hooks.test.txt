

  Connection
    #find()
      âœ“ respond with matching records

  Connection (save error)
    #find()
      1) "before each" hook for "respond with matching records"


  1 passing (18ms)
  1 failing

  1) Connection (save error)
       #find()
         "before each" hook for "respond with matching records":
     Error: Save error
      at Suite.<anonymous> (test/fixtures/asynchronous-hooks.test.js:56:18)
      at Object.<anonymous> (test/fixtures/asynchronous-hooks.test.js:50:1)
      at Module._compile (internal/modules/cjs/loader.js:936:30)
      at Object.Module._extensions..js (internal/modules/cjs/loader.js:947:10)
      at Module.load (internal/modules/cjs/loader.js:790:32)
      at Function.Module._load (internal/modules/cjs/loader.js:703:12)
      at Module.require (internal/modules/cjs/loader.js:830:19)
      at require (internal/modules/cjs/helpers.js:68:18)
      at Array.forEach (<anonymous>)
      at Module._compile (internal/modules/cjs/loader.js:936:30)
      at Object.Module._extensions..js (internal/modules/cjs/loader.js:947:10)
      at Module.load (internal/modules/cjs/loader.js:790:32)
      at Function.Module._load (internal/modules/cjs/loader.js:703:12)
      at Function.Module.runMain (internal/modules/cjs/loader.js:999:10)
      at /Users/ryusi/.nodebrew/node/v12.10.0/lib/node_modules/npm/node_modules/libnpx/index.js:268:14



